### ISSUE TITLE
undefined

### ISSUE THREAD

**Issue by Mennaruuk** [freearhey/iptv-checker#24]
Hi. I've been attempting to use iptv-checker on some playlists, but I haven't been able to make it work. It keeps telling me _undefined_. I looked through the output files, and they only have _#EXTM3U_ at the top of the file. Even with URLs, it gives unrealistic outputs. For example, one can test this with https://iptv-org.github.io/iptv/countries/iq.m3u, and there it says all channels are offline.

![image](https://user-images.githubusercontent.com/52135169/133783899-2805d26d-5cd4-461b-8962-f4a7b08d046a.png)

I have FFMPEG installed via Chocolatey and added to system PATH.

Can you please help me with this error? Thank you!

**Comment by freearhey** [comment]
I cannot reproduce this. Here are my results:
![image](https://user-images.githubusercontent.com/7253922/134499500-e98e1490-a395-4bce-a42c-77a5d3b954ca.png)

**Comment by lck1115** [comment]
all offline here. what's wrong?

**Comment by yadalik** [comment]
I'm getting pretty quick and **false checks** with online file containing only #EXTM3U text, with this undefined thingy at the end of list 
![image](https://user-images.githubusercontent.com/41987955/143545835-a9b76c5c-a765-4a66-9cae-63521138a39b.png)
 for this playlist:
https://smarttvapp.ru/app/iptvfull.m3u (downloaded local) on
**Windows 10** (20H2 19042.1348)
npm lts -version (8.1.4, nvm ls: * 16.13.0 (Currently using 64-bit executable))
ffmpeg build 2021-11-25-git-522f577d7e-full_build-www.gyan.dev, with binaries in PATH.
Maybe it has some errors on connecting ffmpeg? Cause it pass through ~500 channels way too quickly (with default concurrency in ~50s).

**But** when I run this on **WSL2** (ffmpeg installed from apt get) - everything **works fine** (and slow, but with -p 30 pretty good). But also shows this undefined thing at the end: 
![image](https://user-images.githubusercontent.com/41987955/143545168-815100bd-6919-422e-8183-19bb67cdb3bf.png)

**Comment by connord96** [comment]
I'm getting the same results as yadalik above:
![image](https://user-images.githubusercontent.com/33485973/145503413-15f32de7-7068-4131-9268-75c3a7461269.png)

I am also running similar specs.

**Comment by freearhey** [comment]
Try to run this command:

```
ffprobe -of json -hide_banner -timeout '60000000' 'https://live-hls-web-aje.getaj.net/AJE/index.m3u8'
```

This is essentially the command that is run under the hood when checking the streams. 

If all goes well, you should see something similar in the console:

<details>
<summary>Log</summary>

```
{
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:6')
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-INDEPENDENT-SEGMENTS')
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/01.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/01.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE/02.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/02.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE/03.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/03.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE/04.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/04.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE/05.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/05.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE/06.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[https @ 0x7fc1c4026000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/06.m3u8' for reading
[hls @ 0x7fc1c300e000] Skip ('#EXT-X-VERSION:3')
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-01-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-01-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-01-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-01-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-02-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-02-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-02-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-02-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-03-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-03-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-03-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-03-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-04-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-04-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-04-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-04-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-05-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-05-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-05-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-05-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-06-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE/20201001T013811-06-3862637.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-06-3862636.ts' for reading
[hls @ 0x7fc1c300e000] Opening 'https://live-hls-web-aje.getaj.net/AJE-b/20201001T013811-06-3862637.ts' for reading
[h264 @ 0x7fc1c4052a00] Increasing reorder buffer to 2
[h264 @ 0x7fc1c38ad200] Increasing reorder buffer to 2
[h264 @ 0x7fc1c503a600] Increasing reorder buffer to 2
[h264 @ 0x7fc1c38d6e00] Increasing reorder buffer to 2
[h264 @ 0x7fc1c38e8c00] Increasing reorder buffer to 2
[h264 @ 0x7fc1c40bb600] Increasing reorder buffer to 2
[h264 @ 0x7fc1c40c2600] Increasing reorder buffer to 2
[h264 @ 0x7fc1c40ee200] Increasing reorder buffer to 2
[h264 @ 0x7fc1c30b1600] Increasing reorder buffer to 2
[h264 @ 0x7fc1c30ce800] Increasing reorder buffer to 2
[h264 @ 0x7fc1c397c800] Increasing reorder buffer to 2
[h264 @ 0x7fc1c70ed000] Increasing reorder buffer to 2
Input #0, hls, from 'https://live-hls-web-aje.getaj.net/AJE/index.m3u8':
  Duration: N/A, start: 50524.787133, bitrate: N/A
  Program 0 
    Metadata:
      variant_bitrate : 6324320
  Stream #0:0: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 1920x1080 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 6324320
  Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 6324320
  Program 1 
    Metadata:
      variant_bitrate : 6324320
  Stream #0:2: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 1920x1080 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 6324320
  Stream #0:3: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 6324320
  Program 2 
    Metadata:
      variant_bitrate : 3120800
  Stream #0:4: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 1280x720 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 3120800
  Stream #0:5: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 3120800
  Program 3 
    Metadata:
      variant_bitrate : 3120800
  Stream #0:6: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 1280x720 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 3120800
  Stream #0:7: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 3120800
  Program 4 
    Metadata:
      variant_bitrate : 1831872
  Stream #0:8: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 960x540 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 1831872
  Stream #0:9: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 1831872
  Program 5 
    Metadata:
      variant_bitrate : 1831872
  Stream #0:10: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 960x540 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 1831872
  Stream #0:11: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 1831872
  Program 6 
    Metadata:
      variant_bitrate : 1013696
  Stream #0:12: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 746x420 [SAR 1120:1119 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 1013696
  Stream #0:13: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 1013696
  Program 7 
    Metadata:
      variant_bitrate : 1013696
  Stream #0:14: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 746x420 [SAR 1120:1119 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 1013696
  Stream #0:15: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 1013696
  Program 8 
    Metadata:
      variant_bitrate : 813664
  Stream #0:16: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 640x360 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 813664
  Stream #0:17: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 813664
  Program 9 
    Metadata:
      variant_bitrate : 813664
  Stream #0:18: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 640x360 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 813664
  Stream #0:19: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 813664
  Program 10 
    Metadata:
      variant_bitrate : 428640
  Stream #0:20: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 426x240 [SAR 640:639 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 428640
  Stream #0:21: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 428640
  Program 11 
    Metadata:
      variant_bitrate : 428640
  Stream #0:22: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv), 426x240 [SAR 640:639 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 428640
  Stream #0:23: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 428640

}
```
</details>

Otherwise, the problem is in the `ffmpeg` library.

**Comment by Hulkaveli** [comment]
same issue here. running on windows it just gets bunk output from ffprobe because the command sent to ffprobe uses single quotes.

the ffprobe command even fails if ran manually unless you replace the single quotes with double quotes
[output.txt](https://github.com/freearhey/iptv-checker/files/7823677/output.txt)
[ffprobe.txt](https://github.com/freearhey/iptv-checker/files/7823678/ffprobe.txt)

**Comment by Hulkaveli** [comment]
i edited the helper.js after i installed it.
made these changes and it looks like its working for me now.

`function buildCommand(item, config) {
  const userAgent = item.http['user-agent'] || config.userAgent
  let args = [
    `ffprobe`,
    `-of json`,
    `-v debug`,
    `-hide_banner`,
    `-show_streams`,
    `-show_format`,
  ]

  if (config.timeout) {
    args.push(`-timeout`, `${config.timeout * 1000}`)
  }

  if (item.http.referrer) {
    args.push(`-headers`, `Referer: "${item.http.referrer}"`)
  }

  if (userAgent) {
    args.push(`-user_agent`, `"${userAgent}"`)
  }

  args.push(`"${item.url}"`)

  args = args.join(` `)

  return args`

**Comment by freearhey** [comment]
Fixed in [v0.23.0](https://github.com/freearhey/iptv-checker/releases/tag/v0.23.0)
